articles/css-flexbox-vs-css-grid/index.md

```diff
@@ -0,0 +1,748 @@
+
+On the other hand, a Grid is a two-dimensional layout, meaning you can lay out items in rows and columns.
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid.png)
+
+An example of a basic two-dimensional layout.
+
+```html
+<div class="container">
+    <div class="item item1"></div>
+    <div class="item item2"></div>
+    <div class="item item3"></div>
+    <div class="item item4"></div>
+    <div class="item item5"></div>
+    <div class="item item6"></div>
+</div>
+```
+
+```css
+.container {
+    display: grid;
+    grid-template-columns: 150px 150px 150px;
+    grid-gap: 1rem;
+    grid-template-rows: 150px 150px;
+}
+
+.item {
+    background-color: #1EAAFC;
+    color: #fff;
+    border: 6px solid #171717;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid-example.jpg)
+
+We specify a Grid with `display: grid` and then determine rows and columns with `grid-template-columns` and `grid-template-rows`. This defines a two-dimensional layout. In this case, each column and row take the size of `150px`.
+
+#### Content vs layout
+
+Flexbox is content-based. Content comes first. By content first means it help you align contents and move blocks and items within a container.
+
+On the other hand, the Grid focuses on laying out the outer/skeleton layout of the page.
+
+To understand the content vs layout concept, let's go back to the header navigation bar design we've discussed above. We can target an item inside the header and determine where to display it.
+
+To get the core difference, Assuming we want to move the `SIGH IN` and `SIGH UP` menu to the left, we can do it using the Flex and Grid.
+
+Using flex;
+
+```css
+.nav {
+    display: flex;
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 1px;
+}
+
+.nav div {
+    margin-left: 20px;
+    margin-right: 20px;
+    color: whitesmoke;
+}
+
+di v:nth-child(4) {
+    margin-left: auto;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-content-first.png)
+
+We specify the child item of the container and determine its position. That’s what we mean by referring to flex as content-based.
+
+On the other hand, we can implement the same using the Grid. The header is one dimensional, and Flexbox is perfect for doing this, but we can do that with the Grid to differentiate between layout and content.
+
+```css
+.nav-grid {
+    display: grid;
+    grid-template-columns: repeat(10, 1fr);
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 1px;
+}
+
+.nav-grid div {
+    margin-left: 20px;
+    margin-right: 20px;
+    color: whitesmoke;
+}
+```
+
+And the result looks pretty the same.
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid-header-nav.png)
+
+But notice, we specify ten columns, each taking a fraction of one unit. To drive the point home, let's see how the Grid layout works under the hood.
+
+Use chrome to inspect the parent `div` element.
```

Comment:
Capitalise Chrome (though you could just say web browser since all them have an Inspector tool)